File: Java/Q1TwoSum.java
/*
@b-knd (jingru) on 20 July 2022 09:47:00
*/

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

class Q1TwoSum {
    public int[] twoSum(int[] nums, int target) {
        List<Integer> list = Arrays.stream(nums).boxed().collect(Collectors.toList());
        int[] res = new int[2];
        for(int i = 0; i <= list.size(); i++){
            int complement = target - list.get(i);
            if(list.contains(complement) && i != list.indexOf(complement)){
                res[0] = i;
                res[1] = list.indexOf(complement);
                break;
            }
        }
        return res;
    }
}

File: Python/Q1TwoSum.py
# @b-knd (jingru) on 07 August 2022 14:49:00

class Solution(object):
    def twoSum(self, nums, target):
        complements = {};

        for i, value in enumerate(nums):
            complement = target - nums[i]
            if complement in complements:
                return [i, complements[complement]]
            else:
                complements[value] = i

    
#Runtime: 37 ms, faster than 99.07% of Python online submissions for Two Sum.
#Memory Usage: 14.5 MB, less than 22.61% of Python online submissions for Two Sum.
